recipes/imds_config.rb

```diff
@@ -49,3 +49,7 @@
     raise "head_node_imds_secured must be 'true' or 'false', but got #{node['cluster']['head_node_imds_secured']}"
   end
 end
+
+# IMDS access must be verified on cluster creation, but skipped on cluster update
+# because customer may have introduced IMDS access customizations.
+include_recipe 'aws-parallelcluster::test_imds' unless node['conditions']['on_update'] == 'true'
```

Comment:
the test_imds recipe relies on the get_system_users library function, which is defined as:

```ruby
def get_system_users
  cmd = Mixlib::ShellOut.new("cat /etc/passwd | cut -d: -f1")
  cmd.run_command
  cmd.stdout.split(/\n+/)
end
```

will the assumption of /etc/passwd as the only source of user information be an issue for customers that store it elsewhere?

Reply:
I think it is a correct assumptions at creation time because, at that time, users are managed with the standard way that stores them in `/etc/passwd`.

On update time, this could be not the case because customers could have introduced the integration with an external tool, that stores them elsewhere.

I don't know the answer for every tool that customers can use, but with OpsWorks this approach works as new users are stored in `/etc/passwd` wheres with SimpleAD it does not.

In any case, as long as the cluster configuration does not provide a way to manage users with external integrations, we cannot cover all the cases.

My proposal is to limit the check on /etc/passwd.

Reply:
Since, as per comment above, we are going to remove the support for the update, I think it is enough to rely on /etc/passwd